engine/table/src/main/java/io/deephaven/engine/table/impl/SelectOrUpdateListener.java

```diff
@@ -0,0 +1,138 @@
+package io.deephaven.engine.table.impl;
+
+import io.deephaven.engine.rowset.TrackingRowSet;
+import io.deephaven.engine.rowset.WritableRowSet;
+import io.deephaven.engine.table.ModifiedColumnSet;
+import io.deephaven.engine.table.TableUpdate;
+import io.deephaven.engine.table.impl.perf.UpdatePerformanceTracker;
+import io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer;
+import io.deephaven.engine.table.impl.util.AsyncClientErrorNotifier;
+import io.deephaven.engine.updategraph.LogicalClock;
+import io.deephaven.engine.updategraph.TerminalNotification;
+import io.deephaven.engine.updategraph.UpdateGraphProcessor;
+import io.deephaven.engine.util.systemicmarking.SystemicObjectTracker;
+
+import java.io.IOException;
+import java.util.BitSet;
+import java.util.Map;
+
+/**
+ * A Shift-Aware listener for Select or Update. It uses the SelectAndViewAnalyzer to calculate how columns affect other
+ * columns, then creates a column set transformer which will be used by onUpdate to transform updates.
+ */
+class SelectOrUpdateListener extends BaseTable.ListenerImpl {
+    private final QueryTable dependent;
+    private final TrackingRowSet resultIndex;
+    private final ModifiedColumnSet.Transformer transformer;
+    private final SelectAndViewAnalyzer analyzer;
+
+    private long initialNotificationStep = NotificationStepReceiver.NULL_NOTIFICATION_STEP;
+    private long finalNotificationStep = NotificationStepReceiver.NULL_NOTIFICATION_STEP;
```

Comment:
This is more complicated than it needs to be. Use a `boolean updateInProgress` or similar. Set it to one value when work begins, the other when work ends, check for the idle value.